### PR TITLE
Backwards compatible clean with strip

### PR DESCRIPTION
**Pull request by sergiusens**
Automatically switches to call in prime and prints a
deprecation warning.

LP: #1590256

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

File: snapcraft/main.py [hunk 1]
```diff
@@ -228,7 +228,12 @@ def run(args, project_options):
     elif argless_command:
         argless_command()
     elif args['clean']:
-        lifecycle.clean(project_options, args['<part>'], args['--step'])
+        step = args['--step']
+        if args['--step'] == 'strip':
```
Comment (kyrofa) [inline snapcraft/main.py]: Might want to use `step` here.

**Comment by kyrofa** [comment]
Small nit, but :+1:

**Comment by elopio** [comment]
:+1:, minus static checks.

**Comment by sergiusens** [comment]
retest this please

**Pull request merged**